app/charts/shared/chart-helpers.tsx

```diff
@@ -300,6 +306,11 @@ export const useStringVariable = makeUseParsedVariable((x) =>
 export const useTemporalVariable = makeUseParsedVariable((x) =>
   parseDate(`${x}`)
 );
+export const useTemporalEntityVariable = (dimensionValues: DimensionValue[]) =>
+  makeUseParsedVariable((x) => {
+    const value = dimensionValues.find((d) => d.label === x);
+    return parseDate(`${value?.position}`);
+  });
```

Comment:
Nit: I am a bit worried by this as we could potentially iterate over a large array, and it seems like it would trigger a O(n2) behavior for getX (since at each get we iterate over the array of dimension values). We could potentially index the dimension values by label when creating the function so that calls of the function do not need to iterate over the array.